### PR TITLE
v0.2.2-rc3

### LINKED ISSUE #4082: Backport PR `#4072`
The backport bot was unable to open a pull request to backport changes from PR #4072. Please open a PR to backport these changes.

### PR DESCRIPTION
**Pull request by Maan2003**
backport #4072, #4090, #4097
fixes #4082

### CODE REVIEW

**Comment by Maan2003** [comment]
I regenerate databases using 
`just prepare_db_migration_snapshots fedimint-ln-server` and (mint, wallet)

**Comment by Maan2003** [comment]
forgot to update snapshots for fedimint-server :man_facepalming:

File: .envrc
Comment (Maan2003) [inline .envrc:1]: oh wrong file staged

**Review by dpc — DISMISSED** [review]

**Comment by Maan2003** [comment]
lets wait for #4097 backport as well

**Review by justinmoon — APPROVED** [review]

**Pull request merged**